log/term/colorlogger.go

```diff
@@ -0,0 +1,149 @@
+package term
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"sync"
+
+	"github.com/go-kit/kit/log"
+)
+
+// Color is the abstract color, the zero value is the Default.
+type Color uint8
+
+const (
+	NoColor = Color(iota)
+	Red
+	Green
+	Yellow
+	Blue
+	Magenta
+	Cyan
+	White
+	Default
+
+	maxColor
+)
+
+var resetColorBytes = []byte("\x1b[0m")
+var fgColorBytes [][]byte
+var bgColorBytes [][]byte
+
+func init() {
+	for color := NoColor; color < maxColor; color++ {
+		fgColorBytes = append(fgColorBytes, []byte(fmt.Sprintf("\x1b[%dm", 30+color)))
+		bgColorBytes = append(bgColorBytes, []byte(fmt.Sprintf("\x1b[%dm", 40+color)))
+	}
+}
```

Comment:
Then just

```
var
fgColorBytes=[...]byte{"\x1b[30m","\x1b[31m","\x1b[32m","\x1b[33m","\x1b[34m","\x1b[35m","\x1b[36m","\x1b[37m","\x1b[38m","\x1b[39m"}
var
bgColorBytes=[...]byte{"\x1b[40m","\x1b[41m","\x1b[42m","\x1b[43m","\x1b[44m","\x1b[45m","\x1b[46m","\x1b[47m","\x1b[48m","\x1b[49m"}
```

Only needed once.

---
https://onename.com/tgulacsi

2015-08-31 19:41 GMT+02:00 Chris Hines <notifications@github.com>:

> In log/term/colorlogger.go
> <https://github.com/go-kit/kit/pull/116#discussion_r38337051>:
>
> > +	White
> > +	Default
> > +
> > +	maxColor
> > +)
> > +
> > +var resetColorBytes = []byte("\x1b[0m")
> > +var fgColorBytes [][]byte
> > +var bgColorBytes [][]byte
> > +
> > +func init() {
> > +	for color := NoColor; color < maxColor; color++ {
> > +		fgColorBytes = append(fgColorBytes, []byte(fmt.Sprintf("\x1b[%dm", 30+color)))
> > +		bgColorBytes = append(bgColorBytes, []byte(fmt.Sprintf("\x1b[%dm", 40+color)))
> > +	}
> > +}
>
> @tgulacsi <https://github.com/tgulacsi> That's not going to work for me
> as a maintainer of this package because my development environment doesn't
> have sh, echo, sed, or xargs.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/go-kit/kit/pull/116/files#r38337051>.
>